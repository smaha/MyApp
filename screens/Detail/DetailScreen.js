import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { storeData, getData, STORAGE_KEYS } from '../../services/Storage';
import colors from '../../utils/colors';
import BasicButton from '../../components/BasicButton';
import NumberInfo from '../../components/NumberInfo';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  infoArea: {
    flex: 2,
  },
  btn: {
    flex: 3,
  },
});

const DetailScreen = ({ route }) => {
  const { clickNumber } = route.params;
  const [persistedValue, setPersistedValue] = React.useState(null);
  const onSaveState = () => {
    storeData(STORAGE_KEYS.TOTAL, clickNumber);
  };
  React.useEffect(() => {
    getData(STORAGE_KEYS.TOTAL).then((storedValue) => {
      if (storedValue) {
        setPersistedValue(storedValue);
      }
    });
  }, []);
  return (
    <View style={styles.container}>
      <View style={styles.infoArea}>
        <NumberInfo label="Persisted value" value={persistedValue} />
        <NumberInfo label="Current Value" value={clickNumber} />
      </View>
      <View style={styles.btn}>
        <BasicButton
          onPress={onSaveState}
          title="Persist value"
          color={colors.second}
        />
      </View>
    </View>
  );
};

DetailScreen.propTypes = {
  route: PropTypes.object,
};

export default DetailScreen;
