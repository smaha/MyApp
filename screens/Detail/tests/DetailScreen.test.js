import React from 'react';
import { render } from '@testing-library/react-native';
import DetailScreen from '../DetailScreen';

describe('<DetailScreen />', () => {
  test('To Match Snapshot', () => {
    const props = { route: { params: { clickNumber: 5 } } };
    const rended = render(
      <DetailScreen {...props} />,
    );

    expect(rended.toJSON()).toMatchSnapshot();
  });
});
