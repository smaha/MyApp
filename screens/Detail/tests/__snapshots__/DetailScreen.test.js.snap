// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailScreen /> To Match Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "textAlign": "center",
          }
        }
      >
        Persisted value
      </Text>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 0.5,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "flex": 1,
            "textAlign": "center",
          }
        }
      >
        Current Value
      </Text>
      <Text
        style={
          Object {
            "flex": 1,
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#ffba93",
              },
            ]
          }
        >
          Persist value
        </Text>
      </View>
    </View>
  </View>
</View>
`;
